src/MPAGSCipher/VigenereCipher.cpp

```diff
@@ -0,0 +1,88 @@
+#include "VigenereCipher.hpp"
+#include "Alphabet.hpp"
+#include"CaesarCipher.hpp"
+
+#include <algorithm>
+#include <string>
+#include <iostream>
+
+VigenereCipher::VigenereCipher(const std::string& key)
+{
+    // Set the given key
+    setKey(key);
+}
+
+void VigenereCipher::setKey(const std::string& key)
+{
+    key_ = key;
+
+    // Make sure the key is uppercase
+
+    std::transform(std::begin(key_), std::end(key_), std::begin(key_), ::toupper);
+
+    // Remove non-alphabet characters
+
+    key_.erase(std::remove_if(std::begin(key_), std::end(key_),
+                              [](char c) { return !std::isalpha(c); }),
+               std::end(key_));
+
+    // Check if the key is empty and replace with default if so
+
+    if (key_ == ""){
+        key_= "DEFAULT";
+    }
+
+    keyLength_ = key_.size();
+    charLookup_.clear();
+
+    // loop over the key
+    
+    for (const auto& keyChar : key_) {
+        //std::size_t charPos {};
+        // For each character in the key, find the corresponding position in
+        // the alphabet by using an indexed loop over the alphabet container
+        // for (std::size_t i{0}; i < Alphabet::size; ++i) {
+        //     if (keyChar == Alphabet::alphabet[i]) {
+        //         charPos = i ;
+        //         break;
+        //     }
+        //}
+
+        const std::size_t charPos {Alphabet::alphabet.find(keyChar)};
+
+        //std::cout << charPos <<std::endl;
+        // Create a CaesarCipher using this position as a key
+        CaesarCipher cc{charPos};
+
+        // Insert a std::pair of the letter and CaesarCipher into the lookup
+        auto pairToAdd {std::make_pair(keyChar, cc)};
+        charLookup_.insert(pairToAdd);
```

Comment:
Can simplify this slightly:
```suggestion
        charLookup_.insert( std::make_pair(keyChar, cc) );
```